src/middlewared/middlewared/service/crud_service.py

```diff
@@ -152,9 +152,8 @@ async def query(self, filters, options):
         # for filters for performance reasons.
         if not options['force_sql_filters'] and options['extend']:
             datastore_options = options.copy()
-            datastore_options.pop('count', None)
-            datastore_options.pop('get', None)
-            datastore_options.pop('select', None)
+            for option in ['count', 'get', 'limit', 'offset', 'select']:
```

Comment:
Should we make a global object like
```
PAGINATION_OPTS = ('count', 'get', 'limit', 'offset', 'select')
```

And use it in this loop instead? Ideally the global variable might help us catch this in the future? Furthermore, we should probably add some tests for these.